src/component/handlers/edit/editOnInput.js

```diff
@@ -110,7 +119,22 @@ function editOnInput(editor: DraftEditor): void {
     // This can be buggy for some Android keyboards because they don't fire
     // standard onkeydown/pressed events and only fired editOnInput
     // so domText is already changed by the browser and ends up being equal
-    // to modelText unexpectedly
+    // to modelText unexpectedly.
+    // Newest versions of Android support the dom-inputevent-inputtype
+    // and we can use the `inputType` to properly apply the state changes.
+
+    /* $FlowFixMe inputType is only defined on a standard sadraft of a standard.
+     * https://w3c.github.io/input-events/#dom-inputevent-inputtype */
+    const {inputType} = e.nativeEvent;
+    // TODO add check for "input events level 2" https://w3c.github.io/input-events/#dom-inputevent-inputtype
+    if (inputType) {
+      let newEditorState = onInputType(inputType, editorState);
+      if (newEditorState !== editorState) {
+        editor.restoreEditorDOM();
+        editor.update(newEditorState);
+        return;
+      }
+    }
```

Comment:
This change is necessary because `keyDown` on draft-js `edit` mode on Android doesn't come with the correct `keyCode`, so, for example, when users press `backspace` we are not actually executing `keyCommandPlainBackspace`, which leads to an incorrect content state.

The input event comes with this `inputType` which lets us execute the correct action.
I still need to make sure older versions of Android also support this property (will do soon).